### PR TITLE
Basic notes field for VM and node

### PR DESCRIPTION
**Pull request by ricco386**
Closes the first phase of issue https://github.com/erigones/esdc-ce/issues/98#issuecomment-313396929

Once merged set #98 to milestone 2.7.0

### CODE REVIEW

File: api/node/define/views.py [hunk 1]
```diff
@@ -71,6 +71,8 @@ def node_define(request, hostname, data=None):
         :type data.is_compute: boolean
         :arg data.is_backup: Backup capability
         :type data.is_backup: boolean
+        :arg data.note: Text note visible to every user with access to node.
```
Comment (dn0) [inline api/node/define/views.py]: No dot at end of this sentence :)

File: api/vm/define/views.py [hunk 2]
```diff
@@ -160,6 +162,8 @@ def vm_define(request, hostname_or_uuid, data=None):
         :type data.vcpus: integer
         :arg data.ram: Size of RAM inside VM (32 - 524288 MB)
         :type data.ram: integer
+        :arg data.note: Text note visible to every user with access to VM.
```
Comment (dn0) [inline api/vm/define/views.py]: No dot at end of sentence :)

File: api/vm/define/views.py [hunk 3]
```diff
@@ -97,6 +97,8 @@ def vm_define(request, hostname_or_uuid, data=None):
         :type data.vcpus: integer
         :arg data.ram: **required** - Size of RAM inside VM (32 - 524288 MB)
         :type data.ram: integer
+        :arg data.note: Text note visible to every user with access to VM.
```
Comment (dn0) [inline api/vm/define/views.py]: No dot at end of sentence and for post there should be a default value mentioned in brackets.

File: gui/templates/gui/node/details.html [hunk 4]
```diff
@@ -153,6 +153,12 @@
             <td>{% trans "Status" %}:</td>
             <td><span class="label status_node{{ node.status }}" id="node_label_{{ node.hostname }}">{{ node.get_status_display }}</span></td>
           </tr>
+          <tr>
+            <td>{% trans "Note" %}:</td>
+            <td>
+                <span class="node_note">{{ node.note }}</span>
```
Comment (dn0) [inline gui/templates/gui/node/details.html]: Strange CSS class, consider using `vm_hostname` CSS class for smaller/dynamic text size.

File: gui/templates/gui/vm/details.html [hunk 5]
```diff
@@ -367,6 +367,12 @@
                 </td>
               {% endwith %}
             </tr>
+            <tr>
+              <td>{% trans "Note" %}:</td>
+              <td>
+                  <span class="vm_note">{{ vm.note }}</span>
```
Comment (dn0) [inline gui/templates/gui/vm/details.html]: Strange CSS class, consider using `vm_hostname` CSS class for smaller/dynamic text size.

**Comment by dn0** [comment]
I'm wondering, what happens if I do: `es set /vm/<hostname>/define -note null`?

**Comment by ricco386** [comment]
Both `es set /vm/<hostname>/define -note null` and `es set /node/<hostname>/define -note null` store note as empty string `"note": ""`

**Pull request merged**